### PR TITLE
MAYA-125028 add layer muting serialization to the proxy shape

### PR DESCRIPTION
**Pull request by pierrebai-adsk**
- Refactor common code to manage a dynamic string attribute on a Maya node.
- Refactor the load rules to use the new common dynamic attribute code.
- Add a new attribute on the proxy shape for layer muting.
- Added unit tests.

### CODE REVIEW

File: lib/mayaUsd/utils/layerMuting.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+namespace MAYAUSD_NS_DEF {
+
+// The muted state of a layer is stage-level data. As such, it is not saved
+// within the layer (i.e. in the USD files that have been staged.) The reason
+// behind this is that two stages could have different muted layers, a single
+// layer could be muted in one stage and not muted in another stage. So, the
+// muted state cannot be a layer-level data.
+//
+// Furthermore, stages in USD are not saved but are apure-run-time entity,
```
Comment (pierrebai-adsk) [inline lib/mayaUsd/utils/layerMuting.h:34]: typos: space missing and remove dashes

File: lib/mayaUsd/utils/layerMuting.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+// limitations under the License.
+//
+#ifndef MAYAUSD_LAYERMUTING_H
+#define MAYAUSD_LAYERMUTING_H
+
+#include <mayaUsd/base/api.h>
+
+#include <pxr/usd/usd/stage.h>
+
+#include <maya/MObject.h>
```
Comment (pierrebai-adsk) [inline lib/mayaUsd/utils/layerMuting.h:23]: Replace with MApiNamespace.h

File: lib/mayaUsd/utils/layerMuting.cpp [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+// to ensure we don't get any issue and can select the separator used to split
+// identifiers as we wish.
+
+MString convertIdentifierToText(const std::string& identifier)
+{
+    // Note: do *not* use MString to concatenate as it treat single char as int!
+    std::string text;
+
+    for (const char c : identifier) {
+        const char c1 = 'A' + (((unsigned int)c) / 16u);
```
Comment (pierrebai-adsk) [inline lib/mayaUsd/utils/layerMuting.cpp:47]: Is this encoding really needed?

File: lib/mayaUsd/utils/layerMuting.cpp [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+MString convertLayerMutingToText(const PXR_NS::UsdStage& stage)
+{
+    MString text;
+
+    std::vector<std::string> muted = stage.GetMutedLayers();
+    for (const std::string& item : muted) {
+        if (text.length() > 0) {
+            text += "/";
```
Comment (pierrebai-adsk) [inline lib/mayaUsd/utils/layerMuting.cpp:112]: Use string arrays instead of a special separator.

**Review by ppt-adsk — COMMENTED** [review]
Looks good!  Just a trivial const ref argument change to make.

**Review by ppt-adsk — APPROVED** [review]

**Comment by pierrebai-adsk** [comment]
Only PF failure are the two Linux interactive builds failed to launch... unrelated.

**Comment by seando-adsk** [comment]
@pierrebai-adsk I removed the "ready to merge" label. The cot team is currently working on our Linux GPU machines trying to fix them. That is why your preflight had errors. Once the machines are fixed you'll need to re-run the preflight. Please wait to do that until I announce the machines are fixed.

**Pull request merged**